internal/transport/transport_test.go

```diff
@@ -461,40 +461,151 @@ func setUpWithOptions(t *testing.T, port int, sc *ServerConfig, ht hType, copts
 	return server, ct.(*http2Client), cancel
 }
 
-func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.Conn) (*http2Client, func()) {
-	lis, err := net.Listen("tcp", "localhost:0")
+// A server which can be switched from responsive to non-responsive by writing a toggle boolean to the respModeCh channel
+// if a respModeCh channel is created.
+type pingModeServer struct {
+	respModeCh chan bool
+	error      chan error
+	ready      bool
+	cancel     context.CancelFunc
+	conn       net.Conn
+	lis        net.Listener
+}
+
+func (s *pingModeServer) start(t *testing.T) {
+	conn, err := s.lis.Accept()
 	if err != nil {
-		t.Fatalf("Failed to listen: %v", err)
+		s.stop()
+		s.error <- fmt.Errorf("Error at server-side while accepting: %v", err)
+		return
 	}
-	// Launch a non responsive server.
-	go func() {
-		defer lis.Close()
-		conn, err := lis.Accept()
+	s.conn = conn
+	if _, err := io.ReadFull(conn, make([]byte, len(clientPreface))); err != nil {
+		s.stop()
+		s.error <- fmt.Errorf("Error while reading client preface: %v", err)
+		return
+	}
+
+	framer := http2.NewFramer(conn, conn)
+	if err := framer.WriteSettings(); err != nil {
+		s.stop()
+		s.error <- fmt.Errorf("Error at server-side while writing settings: %v", framer.ErrorDetail())
+		return
+	}
+	if err := framer.WriteSettingsAck(); err != nil {
+		s.stop()
+		s.error <- fmt.Errorf("Error while writing settings: %v", err)
+		return
+	}
+	close(s.error)
+	s.ready = true
+	if s.respModeCh != nil {
```

Comment:
I think a comment here about to callout the case if `s.respModeCh == nil` then server does not responToPing